### DISCUSSION TITLE
Repeater field type

### DISCUSSION THREAD

**Discussion by modmatrix** [orchidsoftware/platform#1964] [Ideas]
**Is your feature request related to a problem? Please describe.**

Suppose that a Company model hasMany Locations. Each Location has several fields (address, city, postcode, etc.)

On the edit screen for the Company model, I'd like to be able to manage these related Location records.

I should be able to:

* View and edit the fields for each Location that has already been added to the Company.
* Click an "Add" button to add a new set of Location fields to the form.
* Click a "Delete" button next to an existing Location to remove it.

**Describe the solution you'd like**

I would like a new field type – something like this:

```
Repeater::make('company.locations')
    ->layout([
            Layout::rows([
                Input::make('address'),
                Input::make('city'),
                Input::make('postcode'),
            ])
        ])
```

Perhaps it could have additional methods, such as:

* `max(int $max)`: Specifies the maximum number of relations that can be added.
* `confirmDelete(string $confirmation_prompt)`: Indicates that the user should be prompted for confirmation when deleting a record.

When the form is submitted to the Screen handler, the request data should look like something like this:

```
[
    'company' => [
        'locations' => [
            0 => [
                'id' => 17,
                'address' => '123 Main Street',
                'city' => 'Springfield',
                'postcode' => '12345'
            ],
            1 => [
                'id' => 413,
                'address' => '456 Elm Lane',
                'city' => 'Chicago',
                'postcode' => '67104'
            ],
            2 => [
                'id' => null,
                'address' => '789 Greene Blvd',
                'city' => 'San Francisco',
                'postcode' => '99814'
            ],
        ]
    ]
]
```

The first two locations are examples of records that already existed in the database when the form was rendered. The third location (with a null ID) is an example of a record that was just created in the Repeater field by the user.

There may be better ways of approaching some of these things. This is just the gist of the idea.

**Describe alternatives you've considered**

There's this third-party package:

https://github.com/Nks/orchid-repeater-field

...but I'm hesitant to rely on a package that is so little-used. It seems like this functionality should be included in Orchid.

**Comment by copostic** [comment]
Great, so this is more or less what I ask about here [How to create recursive elements on a form](https://github.com/orchidsoftware/platform/discussions/2124) !
